submodules/power_management_services_setup/main.tf

```diff
@@ -0,0 +1,108 @@
+#####################################################
+# Configure Squid client
+#####################################################
+
+resource "null_resource" "perform_proxy_client_setup" {
+
+  count = var.perform_proxy_client_setup != null ? length(var.perform_proxy_client_setup["squid_client_ips"]) : 0
+
+  connection {
+    type         = "ssh"
+    user         = "root"
+    bastion_host = var.access_host_or_ip
+    host         = var.perform_proxy_client_setup["squid_client_ips"][count.index]
+    private_key  = var.ssh_private_key
+    agent        = false
+    timeout      = "15m"
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+
+      #######  SQUID Forward PROXY CLIENT SETUP ############
+      "LINE=$'export http_proxy=http://${var.perform_proxy_client_setup["squid_server_ip"]}:3128\nexport https_proxy=http://${var.perform_proxy_client_setup["squid_server_ip"]}:3128\nexport HTTP_proxy=http://${var.perform_proxy_client_setup["squid_server_ip"]}:3128\nexport HTTPS_proxy=http://${var.perform_proxy_client_setup["squid_server_ip"]}:3128\nexport no_proxy=${var.perform_proxy_client_setup["no_proxy_env"]}'",
```

Comment:
Could you extract those scripts out as file, and use the script block to reference them? There are static checks on the script file in the pipeline that can help catch issues.

Reply:
would be good. Can you give an example?

Reply:
Yes its good to put it in a script will work on it and push it